### DISCUSSION TITLE
[REJECTED] [struct] set access modifier

### DISCUSSION THREAD

**Discussion by igotfr** [vlang/v#13727] [Ideas]
```v
struct Sct {
set:
  n int
}

s := Sct{n: 5}
s.n = 7 // ok
println(s.n) // error
```

**Comment by igotfr** [comment]
related:
https://github.com/vlang/rfcs/issues/20

**Comment by dumblob** [comment]
What is the motivation? Any cool examples of real-world scenarios?

Btw. how would this behave with "read-update-store" operations like `+=` or any operation which needs to **read** the current value to derive a new value from it?

**Reply by igotfr** [reply]
> What is the motivation? Any cool examples of real-world scenarios?
> 
> Btw. how would this behave with "read-update-store" operations like `+=` or any operation which needs to **read** the current value to derive a new value from it?

a property that can't be accessed directly, only setted, but it's useful for calculations in methods with visible return (computed), similar to private, but of course, no inheritance

**Reply by Wajinn** [reply]
I'm confused, and don't understand the intent of the behavior.  Seems like one would just put `mut` at the struct and variable, if wanting the value to change.  Otherwise, assign it just the one time, then it can't be changed in the function, else get an error.

Edit: OP responded, as I was typing a response.

```
struct Sct {
	n int
}
struct Sct2 {
	mut:
	n int
}

s := Sct{n: 7}
mut ss := Sct2{}

// s.n = 8 // will cause an error
ss.n = 7

println(s.n) 
println(ss.n)
```

**Reply by Wajinn** [reply]
> > What is the motivation? Any cool examples of real-world scenarios?
> > Btw. how would this behave with "read-update-store" operations like `+=` or any operation which needs to **read** the current value to derive a new value from it?
> 
> a property that can't be accessed directly, only setted, but it's useful for calculations in methods with visible return (computed), similar to private, but of course, no inheritance

Are we "kind of" going down the "getters" and "setters" OOP path?

**Reply by igotfr** [reply]
> > > What is the motivation? Any cool examples of real-world scenarios?
> > > Btw. how would this behave with "read-update-store" operations like `+=` or any operation which needs to **read** the current value to derive a new value from it?
> > 
> > 
> > a property that can't be accessed directly, only setted, but it's useful for calculations in methods with visible return (computed), similar to private, but of course, no inheritance
> 
> Are we "kind of" going down the "getters" and "setters" OOP path?

yes, I know abstract motivation but I don't remember any real-world scenarios

In V, specifically for properties we have:
| | | |
|:---:|:---:|:---:|
| mut | get | set |
| default | get |

with set, would be:
| | | |
|:---:|:---:|:---:|
| set | | set |

**Reply by JalonSolov** [reply]
V is not OOP, and (hopefully) never will be.  Getters and setters are often useless obfuscation for simply setting a variable.  If you want to make a more complicated setter for something (in case you want to validate values being set are within a specific range, etc.), just make a receiver method yourself.
